lib/ansible/modules/cloud/amazon/ec2_import_snapshot.py

```diff
@@ -0,0 +1,250 @@
+
+    params = {
+        'ImportTaskIds': [import_task_id]
+    }
+    start_time = time.time()
+    while True:
+        status = connection.describe_import_snapshot_tasks(**params)
+
+        # What are the valid status values?
+        if len(status['ImportSnapshotTasks']) > 1:
+            raise RuntimeError("Should only be 1 Import Snapshot Task with this id.")
+
+        task = status['ImportSnapshotTasks'][0]
+        if task['SnapshotTaskDetail']['Status'] in ['completed']:
+            return status
+
+        if time.time() - start_time > wait_timeout:
+            raise RuntimeError('Wait timeout exceeded (%s sec)' % wait_timeout)
+
+        time.sleep(15)
+
+
+def import_snapshot(module, connection):
+    description = module.params.get('description')
+    image_format = module.params.get('format')
+    url = module.params.get('url')
+    s3_bucket = module.params.get('s3_bucket')
+    s3_key = module.params.get('s3_key')
+    encrypted = module.params.get('encrypted')
+    kms_key_id = module.params.get('kms_key_id')
+    role_name = module.params.get('role_name')
+    wait = module.params.get('wait')
+    wait_timeout = module.params.get('wait_timeout')
+
+    if not url and not (s3_bucket and s3_key):
+        module.fail_json(msg='url or s3_bucket and s3key are required.')
+    if all([url, s3_bucket, s3_key]):
+        module.fail_json(msg='url or s3_bucket and s3key are required, not all 3.')
+    if not image_format:
+        module.fail_json(msg='format is required.')
+
+    if module.check_mode:
+        module.exit_json(changed=True, msg="IMPORT operation skipped - running in check mode")
+
+    try:
+        params = {
+            'Description': description,
+            'DiskContainer': {
+                'Description': description,
+                'Format': image_format,
+            },
+            'Encrypted': encrypted
+        }
+        if url:
+            params['DiskContainer']['Url'] = url
+        else:
+            params['DiskContainer']['UserBucket'] = {
+                'S3Bucket': s3_bucket,
+                'S3Key': s3_key
+            }
+        if kms_key_id:
+            params['KmsKeyId'] = kms_key_id
+        if role_name:
+            params['RoleName'] = role_name
+
+        task = connection.import_snapshot(**params)
+        import_task_id = task['ImportTaskId']
+        detail = task['SnapshotTaskDetail']
+
+        if wait:
+            status = wait_for_import_snapshot(connection, wait_timeout, import_task_id)
+            detail = status['ImportSnapshotTasks'][0]['SnapshotTaskDetail']
+
+        # common return value for exit_json
+    except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:
+        module.fail_json_aws(e, msg="Error importing image")
+
+    module.exit_json(changed=True, **camel_dict_to_snake_dict(detail))
+
+
+def import_snapshot_ansible_module():
+    argument_spec = ec2_argument_spec()
```

Comment:
This is unnecessary; AnsibleAWSModule automatically merges the argument spec that you pass in with the core EC2 arguments.